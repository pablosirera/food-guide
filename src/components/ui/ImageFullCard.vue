<script setup>
defineProps({
  item: {
    type: Object,
    default: () => ({}),
  },
})
</script>

<template>
  <div class="card w-40 h-40 m-3 shadow-xl image-full">
    <figure>
      <img :src="item.img" />
    </figure>
    <div class="card-body justify-center">
      <h2 class="card-title text-center justify-center">{{ item.name }}</h2>
    </div>
  </div>
</template>
