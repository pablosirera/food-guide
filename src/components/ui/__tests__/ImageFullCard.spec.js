import { mount } from '@vue/test-utils'
import ImageFullCard from '@/components/ui/ImageFullCard.vue'
import { describe, expect, it } from 'vitest'

describe('ImageFullCard.spec.js', () => {
  it('renders', () => {
    const wrapper = mount(ImageFullCard)
    expect(wrapper.exists).toBeTruthy()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should matched', () => {
    const wrapper = mount(ImageFullCard, {
      props: {
        item: {
          img: 'https://api.lorem.space/image/burger?w=400&h=225',
          name: 'Burger King',
        },
      },
    })

    expect(wrapper.text()).toContain('Burger King')
  })
})
