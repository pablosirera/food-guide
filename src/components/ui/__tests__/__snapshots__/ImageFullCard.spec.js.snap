// Vitest Snapshot v1

exports[`ImageFullCard.spec.js > renders 1`] = `
"<div class=\\"card m-3 image-full cursor-pointer normal\\" data-v-05031e22=\\"\\">
  <figure class=\\"w-full h-full overflow-hidden\\" data-v-05031e22=\\"\\"><img lazy=\\"\\" data-v-05031e22=\\"\\" /></figure>
  <div class=\\"card-body justify-center\\" data-v-05031e22=\\"\\">
    <h2 class=\\"card-title text-center justify-center\\" data-v-05031e22=\\"\\"></h2>
  </div>
</div>"
`;
